src/Compilers/CSharp/Portable/Compilation/SyntaxTreeSemanticModel.cs

```diff
@@ -1774,20 +1774,13 @@ private Symbol GetDeclaredMember(NamespaceOrTypeSymbol container, TextSpan decla
                     }
                 }
 
-                foreach (var loc in symbol.Locations)
+                if (symbol.HasLocationContainedWithin(this.SyntaxTree, declarationSpan, out var wasZeroWidthMatch))
```

Comment:
note: This is distinctly different from IsDefinedInSourceTree which does an intersects check, while this helper does a *contains* check.  The intersects check means that nodes *larger than* the decl-span passed in are considered matches.